modular_skyrat/modules/modular_implants/code/soulcatcher/soulcatcher_component.dm

```diff
@@ -0,0 +1,273 @@
+		to_chat(target_nif.linked_mob, message_to_recieve)
+		return TRUE
+
+	return FALSE
+
+/datum/soulcatcher_room
+	/// What is the name of the room?
+	var/name = "Default Room"
+	/// What is the description of the room?
+	var/room_description = "An orange platform suspended in space orbited by reflective cubes of various sizes. There really isn't much here at the moment."
+	/// What souls are currently inside of the room?
+	var/list/current_souls = list()
+	/// Weakref for the master soulcatcher datum
+	var/datum/weakref/master_soulcatcher
+	/// What is the name of the person sending the messages?
+	var/outside_voice = "Host"
+	/// Can the room be joined at all?
+	var/joinable = TRUE
+
+/// Attemps to add a ghost to the soulcatcher room.
+/datum/soulcatcher_room/proc/add_soul_from_ghost(mob/dead/observer/ghost)
+	if(!ghost || !ghost.ckey)
+		return FALSE
+
+	if(!ghost.mind)
+		ghost.mind = new /datum/mind(ghost.key)
+		ghost.mind.name = ghost.name
+		ghost.mind.active = TRUE
+
+	if(!add_soul(ghost.mind))
+		return FALSE
+
+	return TRUE
+
+/// Converts a mind into a soul and adds the resulting soul to the room.
+/datum/soulcatcher_room/proc/add_soul(datum/mind/mind_to_add)
+	if(!mind_to_add)
+		return FALSE
+
+	var/datum/component/soulcatcher/parent_soulcatcher = master_soulcatcher.resolve()
+	var/datum/parent_object = parent_soulcatcher.parent
+	if(!parent_object)
+		return FALSE
+
+	var/mob/living/soulcatcher_soul/new_soul = new(parent_object)
+	new_soul.name = mind_to_add.name
+
+	if(mind_to_add.current)
+		var/datum/component/previous_body/body_component = mind_to_add.current.AddComponent(/datum/component/previous_body)
+		body_component.soulcatcher_soul = WEAKREF(new_soul)
+
+		new_soul.round_participant = TRUE
+		new_soul.body_scan_needed = TRUE
+
+		new_soul.previous_body = WEAKREF(mind_to_add.current)
+		new_soul.name = pick(GLOB.last_names) //Until the body is discovered, the soul is a new person.
+		new_soul.soul_desc = "[new_soul] lacks a discernible form."
+
+	mind_to_add.transfer_to(new_soul, TRUE)
+	current_souls += new_soul
+	new_soul.current_room = WEAKREF(src)
+
+	var/datum/preferences/preferences = new_soul.client?.prefs
+	if(preferences)
+		new_soul.ooc_notes = preferences.read_preference(/datum/preference/text/ooc_notes)
+		if(!new_soul.body_scan_needed)
+			new_soul.soul_desc = preferences.read_preference(/datum/preference/text/flavor_text)
+
+	to_chat(new_soul, span_cyan("You find yourself now inside of: [name]"))
+	to_chat(new_soul, span_notice(room_description))
+	to_chat(new_soul, span_doyourjobidiot("You have entered a soulcatcher, do not share any information you have received while a ghost. If you have died within the round, you do not know your identity until your body has been scanned, standard blackout policy also applies."))
+	log_admin("[new_soul] entered the soulcatcher room, [src]")
+
+	return TRUE
+
+/// Removes a soul from a soulcatcher room, leaving it as a ghost.
```

Comment:
For documentation like this, you should say when it returns true and when it returns false (because I wouldn't expect it to return something intuitively).

Can be as simple as `Returns TRUE if successful, FALSE otherwise.`
This applies to all the other ones that behave like this on this page, and possibly in other files too.